athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/AwsRestHighLevelClient.java

```diff
@@ -94,6 +98,35 @@ public AwsRestHighLevelClient(RestClientBuilder builder)
         return (LinkedHashMap<String, Object>) mappingsResponse.mappings().get(index).sourceAsMap();
     }
 
+    /**
+     * Retrieves cluster-health information for shards associated with the specified index. The request will time out
+     * if no results are returned after a period of time indicated by timeout.
+     * @param index is used to restrict the request to a specified index.
+     * @param timeout is the command timeout period in seconds.
+     * @return a set of shard ids for the specified index.
+     * @throws IOException if an error occurs while sending the request to the Elasticsearch instance.
+     * @throws RuntimeException if the request times out, or no active-primary shards are present.
+     */
+    public Set<Integer> getShardIds(String index, long timeout)
+            throws RuntimeException, IOException
+    {
+        ClusterHealthRequest request = new ClusterHealthRequest(index)
+                .timeout(new TimeValue(timeout, TimeUnit.SECONDS));
+        // Set request to shard-level details
+        request.level(ClusterHealthRequest.Level.SHARDS);
+
+        ClusterHealthResponse response = cluster().health(request, RequestOptions.DEFAULT);
+
+        if (response.isTimedOut()) {
+            throw new RuntimeException("Request timed out.");
+        }
+        else if (response.getActiveShards() == 0) {
+            throw new RuntimeException("There are no active shards.");
```

Comment:
include the index in the error. 

Also, what if a cluster has 5 shards and 1 of the shards is offline and no backup for that shard is available? what detects that situation?

Reply:
Changed error messages.

I'm not sure how we could detect that a specific shard in not available (documentation on that is a bit sketchy). Irrespective, the query for that shard will either throw or time-out. It's better that way, because a shard may only be unavailable temporarily (a short period of time) and may recover in time.